File: question1.js
/**
 * Create something 
 *
 * @param {string} s
 * @param {string} t
 * @returns {string} The string s sorted by occurrence of t
 *
 */
function sortByStrings(s, t){
    let t_map = new Map();

    for (let ch of t) {
        t_map.set(ch, 0);
    }

    for (let ch of s) {
        if (t_map.has(ch)){
            t_map.set(ch, t_map.get(ch) + 1);
        }
    }

    let output = '';
    for (let ch of t_map) {
        output += ch[0].repeat(ch[1]);
    }

    return output;
}

console.log(sortByStrings('weather', 'therapyw'));
console.log(sortByStrings('good', 'odg'));

File: python_soln/question1.py
def sortByStrings(s, t):
    # importing this built-in library is important to ensure that
    # the dictionary remembers the order in which keys are entered
    from collections import OrderedDict

    hash_chars = OrderedDict()
    for char in t:
        hash_chars[char] = 0

    for char in s:
        if hash_chars.get(char) is not None:
            hash_chars[char] += 1

    output = ''
    for char in hash_chars:
        output += hash_chars[char]*char

    return output


print(sortByStrings('weather', 'therapyw'))
print(sortByStrings('good', 'odg'))
